src/roswire/ros2/node.py

```diff
@@ -4,26 +4,50 @@
 import typing
 
 import attr
+import logger
+
+from ..interface import Node
+from .state import ROS2StateProbe
+import dockerblade
 
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
     """Provides an interface for interacting with a ROS2 node.
 
     Attributes
     ----------
     name: str
         The name of the node.
     """
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    node_name: str = attr.ib()
+    _state_probe: 'ROS2StateProbe' = attr.ib(init=False)
+
+    @classmethod
+    def for_app_instance_and_name(cls,
+                                  app_instance: 'AppInstance',
+                                  name: str
+                                  ) -> 'ROS2Node':
+        return ROS2Node(node_name=name, app_instance=app_instance)
+
+    @property
+    def name(self) -> str:
+        return self.node_name
 
     def is_alive(self) -> bool:
         """Determines whether this node is alive."""
-        raise NotImplementedError
+        if self.node_name in self._state_probe.probe().nodes:
+            return True
+        return False
 
     def shutdown(self) -> None:
         """Instructs this node to shutdown."""
-        raise NotImplementedError
+        command = f"ros2 lifecycle set {self.node_name} shutdown"
+        try:
+            self.app_instance.shell.run(command)
+        except dockerblade.exceptions.CalledProcessError:
+            logger.debug(f"Unable to shutdown node {self.node_name}")
```

Comment:
This should probably raise some kind of exception if the node can't be shutdown. We may want to add a new exception for that.

Reply:
I could add an exception for that called `NodeShutdownError` or something like that

Reply:
That sounds great. We might even want to refine the general interface of `shutdown` in `Node` to take an optional keyword argument `ignore_errors` with a default value of `False`. If `ignore_errors` is `True`, we don't raise an exception; if it's `False`, we go ahead and raise a `NodeShutdownError`.